lib/feedsme/index.js

```diff
@@ -54,166 +63,355 @@ function pkgName(pkg) {
  * @api private
  */
-Feedsme.prototype.change = function change(env, data, done) {
-  var pkg = this.extractLatest(data);
-  var app = this.app;
+Feedsme.prototype.change = async function change(env, data) {
+  const pkg = this.extractLatest(data);
+  const { name, version } = pkg;
+
+  const meta = { name, version, env };
+
+  this.log.info(`Processing change for ${ pkgName(pkg) } in ${ env }`, meta);
+  // Update dependent and dependent_of models for reference
+  await this.resolve(env, pkg);
   //
-  // TODO: Figure out if we can get previous here or if we need to do a GET on
-  // artifactory which would kind of suck
+  // Trigger dependent builds and add dependents to parent release line if it
+  // exists
   //
+  await this.trigger(env, data);
 
-  var meta = {
-    name: pkg.name,
-    version: pkg.version,
-    env: env
+  this.log.info(`Successfully processed change for ${ pkgName(pkg) } in ${ env }`, meta);
+};
+
+/**
+ * Trigger potential dependent builds and depOf additions for dependents that
+ * have been published
+ *
+ * @param {String} env Environment we are triggerign for
+ * @param {Object} data full publish payload of package
+ * @returns {Promise} to be resolved
+ */
+Feedsme.prototype.trigger = function trigger(env, data) {
+  const publish = env === 'dev' && data.__published;
+  const pkg = this.extractLatest(data);
+
+  return Promise.all([
+    // Root or Main package triggers dependent builds
+    this.triggerDep({ publish, env, pkg }),
+    // Dependent packages associate with their release-line, only on publish
+    publish && this.triggerDepOf({ pkg })
+  ].filter(Boolean));
+};
+
+/**
+ * Update the releaseline of the given package with the versions of the
+ * dependent builds.
+ *
+ * @param {Object} options Options
+ * @param {Object} options.pkg Package sent to us
+ * @param {Object} options.depOf Parent package lookup if it exists
+ * @returns {Promise} to resolves
+ * @api private
+ */
+Feedsme.prototype.triggerDepOf = async function triggerDepOf({ pkg }) {
+  const { DependentOf } = this.models;
+  const { name, version } = pkg;
+
+  const depOf = await DependentOf.get(name);
+
+  if (!depOf || (depOf && !depOf.dependentOf)) return this.log.info(`${name} does not have a parent package it depends on`);
+
+  this.log.info(`trigger depOf`, depOf.toJSON());
+  // Fetch the release line for the parent package that owns it
+  const depOfName = depOf.dependentOf;
+  let releaseLine = await this.release.get({ pkg: depOfName });
+
+  if (!releaseLine) return this.log.info(`No release line found for ${depOfName}`);
+
+  const rootPkgRange = pkg.dependencies[releaseLine.pkg];
+
+  //
+  // Since we've fetched the latest version we check to see if this dependent
+  // package publish is meant for an older release line
+  //
+  if (!semver.satisfies(releaseLine.version, rootPkgRange)) {
+    releaseLine = await this._walkReleaseLine({
+      condition: (line) => !semver.satisfies(line.version, rootPkgRange),
```

Comment:
nit: You don't technically need the parens here.
```suggestion
      condition: line => !semver.satisfies(line.version, rootPkgRange),
```